services/brig/src/Brig/API/Internal.hs

```diff
@@ -288,19 +289,17 @@ deleteUserNoVerify uid = do
   void $ lift (API.lookupAccount uid) >>= ifNothing userNotFound
   lift $ API.deleteUserNoVerify uid
 
-changeSelfEmailNoSendH :: UserId ::: JsonRequest EmailUpdate -> Handler Response
-changeSelfEmailNoSendH (u ::: req) = do
+changeSelfEmailMaybeSendH :: UserId ::: Bool ::: JsonRequest EmailUpdate -> Handler Response
+changeSelfEmailMaybeSendH (u ::: validate ::: req) = do
   email <- euEmail <$> parseJsonBody req
-  changeSelfEmailNoSend u email >>= \case
+  changeSelfEmailMaybeSend u validate email >>= \case
     ChangeEmailResponseIdempotent -> pure (setStatus status204 empty)
     ChangeEmailResponseNeedsActivation -> pure (setStatus status202 empty)
 
-data ChangeEmailResponse
-  = ChangeEmailResponseIdempotent
-  | ChangeEmailResponseNeedsActivation
-
-changeSelfEmailNoSend :: UserId -> Email -> Handler ChangeEmailResponse
-changeSelfEmailNoSend u email = do
+changeSelfEmailMaybeSend :: UserId -> Bool -> Email -> Handler ChangeEmailResponse
+changeSelfEmailMaybeSend u True email = do
+  API.changeSelfEmail u email
+changeSelfEmailMaybeSend u False email = do
```

Comment:
What does the boolean mean here?

Reply:
20d4a3bc784643976fa319b0074b0b7f38256ffb